### ISSUE TITLE
Implement prod / prod_dim

### ISSUE THREAD

**Issue by unrenormalizable** [tracel-ai/burn#1227]
### Feature description

prod / prod_dim APIs on the lines of torch.prod()

### Feature motivation

For completeness of the API. No concreate use case as of now. My guess is sooner or later some ONNX model we want to import will require this.

### (Optional) Suggest a Solution

PR for the first attempt #1173. That is currently blocked on a number of dependencies not available from the platform layer.

**Comment by antimora** [comment]
Closing this as duplicate of #526
